api/api.py

```diff
@@ -183,14 +183,17 @@ def __init__(self):
         self.response_condition = threading.Condition()
 
     def exchange(self, data: bytes) -> bytes:
```

Comment:
The return type needs to be updated (with `Generator[bytes, None, None]`? cf. https://docs.python.org/3/library/typing.html#typing.Generator)